api/src/main/java/me/lokka30/treasury/api/permission/context/ContextType.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/ArcanePlugins/Treasury>.
+ */
+
+package me.lokka30.treasury.api.permission.context;
+
+import org.jetbrains.annotations.NotNull;
+
+public enum ContextType {
+    GLOBAL,
+    WORLD,
+    SERVER
+    ;
+
+    private static final Context global = new Context(ContextType.GLOBAL, "");
+
+    @NotNull
+    public static Context global() {
+        return global;
+    }
+
+    @NotNull
+    public static Context newContext(@NotNull ContextType type, @NotNull String value) {
+        if (type == GLOBAL) {
+            return global;
+        }
+        if (value.isEmpty()) {
+            throw new IllegalArgumentException("value is empty");
+        }
+        boolean hasSpace = false;
+        for (int i = 0; i < value.length(); i++) {
+            if (Character.isWhitespace(value.charAt(i))) {
+                hasSpace = true;
+                break;
+            }
+        }
+        if (hasSpace) {
+            throw new IllegalArgumentException("Invalid value: value should not contain spaces!");
+        }
```

Comment:
```suggestion
        if (value.indexOf(' ') != -1) {
            throw new IllegalArgumentException("Invalid value: value should not contain spaces!");
        }
```
Better readability and optimized.